### PR TITLE
Reports unused NLS keys and minor misc fixes...

### PR DESCRIPTION
**Pull request by andyHa**
* Using a new job, we can determine unused NLS keys within the application
* Provides some helpers to run code as system tenant
* Prevents an NPE when querying the BlobStorageSpaces without file types
* Skips adding the default port to S3-URLs

### CODE REVIEW

File: src/main/java/sirius/biz/cluster/NLSController.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import sirius.kernel.nls.Translation;
+import sirius.web.controller.BasicController;
+import sirius.web.controller.Routed;
+import sirius.web.http.WebContext;
+import sirius.web.services.JSONStructuredOutput;
+
+/**
+ * Provides services to fetch unused or autocreated NLS keys.
+ * <p>
+ * This is mainly used by {@link sirius.biz.util.ReportUnusedNLSKeysJob} to identify NLS keys which are unused accross
```
Comment (idlira) [inline src/main/java/sirius/biz/cluster/NLSController.java:25]:
```suggestion
 * This is mainly used by {@link sirius.biz.util.ReportUnusedNLSKeysJob} to identify NLS keys which are unused across
```

File: src/main/java/sirius/biz/util/ReportUnusedNLSKeysJob.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                          } else {
+                              missingKeys.retainAll(keysOfNode);
+                          }
+                      });
+
+        if (missingKeys.isEmpty()) {
+            process.log("No unused keys found...");
+        } else {
+            process.log(ProcessLog.info().withFormattedMessage("Exporting %s unused keys...", missingKeys.size()));
+            try (PrintStream out = new PrintStream(process.addFile("unused_keys.txt"))) {
```
Comment (idlira) [inline src/main/java/sirius/biz/util/ReportUnusedNLSKeysJob.java:94]: Why not use process.addTable? One can just visualize or export the table to a local file...

**Review by idlira — CHANGES_REQUESTED** [review]
...

File: src/main/java/sirius/biz/util/ReportUnusedNLSKeysJob.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    @Override
+    public String getDescription() {
+        return "Queries all unused NLS keys on each node in the cluster, computes the intersection and reports this as textfile within the created process.";
+    }
+
+    @Override
+    protected void execute(ProcessContext process) throws Exception {
+        process.log("Fetching unused keys from all nodes...");
+        Set<String> missingKeys = new HashSet<>();
+        Monoflop mf = Monoflop.create();
```
Comment (sabieber) [inline src/main/java/sirius/biz/util/ReportUnusedNLSKeysJob.java:69]: more descriptive name what the monoflip does pls

**Review by sabieber — CHANGES_REQUESTED** [review]

**Review by qw3ry — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

File: src/main/java/sirius/biz/util/ReportUnusedNLSKeysJob.java (hunk 2 above)
Comment (andyHa) [reply src/main/java/sirius/biz/util/ReportUnusedNLSKeysJob.java:100]: This data is processed most probably any way (it is just a single key per line nothing more). Also, running this right after a patch might generate large datasets which will further fille the processlogs table with many more entries.

**Review by idlira — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/util/ReportUnusedNLSKeysJob.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import javax.annotation.Nullable;
+import java.io.PrintStream;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * Collects all unused NLS keys accross all cluster nodes and reports the intersection as text file.
```
Comment (bwiedmann) [inline src/main/java/sirius/biz/util/ReportUnusedNLSKeysJob.java:33]:
```suggestion
 * Collects all unused NLS keys across all cluster nodes and reports the intersection as text file.
```

**Review by bwiedmann — CHANGES_REQUESTED** [review]

**Pull request merged**